### DISCUSSION TITLE
63 characters limit issue with babelfish

### DISCUSSION THREAD

**Discussion by amalbabu-94** [babelfish-for-postgresql/babelfish-for-postgresql#35] [Discussion]
The stored procedure names which I have in my database that contains more than 63 characters get hashed with babelfish.So whenever we create the database scripts using Babelfish,the stored procedure get created with hashed values.As a result when we run the script for creating procedure with an if condition which mentions whether the stored procedure is present or not,the if condition fails as it not able to detect the original stored procedure name(as the hashed name is present as stored procedure name in the database).Is there a way to address this?

**Comment by staticlibs** [comment]
Hi, for procedures the check can be done with `object_id` like this:

```tsql
create procedure very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test
as begin
select 42
end

select * from sys.sysobjects
where id = object_id('very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test')
and objectproperty(id, 'IsProcedure') = 1
```

The mangled `name` column value is a result of the `sys.babelfish_truncate_identifier` function. For index and constraint names the name mangling is a bit different see details in these issues: [1](https://github.com/bill-ramos-rmoswi/babelfish_postgresql_towel/issues/14), [2](https://github.com/bill-ramos-rmoswi/babelfish_postgresql_towel/issues/16). Overall, the checks for mangled names can be run successfully on both Babelfish and MSSQL from the same script (using functions like `IndexExists` from the first link).

**Reply by amalbabu-94** [reply]
Thank you so much,it worked for me.
